packages/cli/src/utils/deploy-v2.ts

```diff
@@ -152,19 +152,54 @@ export async function deploy(mudConfig: MUDConfig, deployConfig: DeployConfig):
   // Register systems (using forEach instead of for..of to avoid blocking on async calls)
   promises = [
     ...promises,
-    ...Object.entries(mudConfig.systems).map(async ([systemName, { fileSelector, openAccess }]) => {
-      console.log(chalk.blue(`Registering system ${systemName} at ${namespace}/${fileSelector}`));
-
-      // Register system at route
-      await fastTxExecute(WorldContract, "registerSystem", [
-        toBytes16(namespace),
-        toBytes16(fileSelector),
-        await contractPromises[systemName],
-        openAccess,
-      ]);
-
-      console.log(chalk.green(`Registered system ${systemName} at ${namespace}/${fileSelector}`));
-    }),
+    ...Object.entries(mudConfig.systems).map(
+      async ([systemName, { fileSelector, openAccess, registerFunctionSelectors }]) => {
+        // Register system at route
+        console.log(chalk.blue(`Registering system ${systemName} at ${namespace}/${fileSelector}`));
+        await fastTxExecute(WorldContract, "registerSystem", [
+          toBytes16(namespace),
+          toBytes16(fileSelector),
+          await contractPromises[systemName],
+          openAccess,
+        ]);
+        console.log(chalk.green(`Registered system ${systemName} at ${namespace}/${fileSelector}`));
+
+        // Register function selectors for the system
+        if (registerFunctionSelectors) {
+          const functionSignatures: FunctionSignature[] = await loadFunctionSignatures(systemName);
+          const isRoot = namespace === "";
+          for (const { functionName, functionArgs } of functionSignatures) {
+            const functionSignature = isRoot
+              ? functionName + functionArgs
+              : `${namespace}_${fileSelector}_${functionName}${functionArgs}`;
+
+            console.log(chalk.blue(`Registering function "${functionSignature}"`));
+            if (isRoot) {
+              const worldFunctionSelector = toFunctionSelector(
+                functionSignature === ""
+                  ? { functionName: systemName, functionArgs } // Register the system's fallback function as `<systemName>(<args>)`
+                  : { functionName, functionArgs }
+              );
```

Comment:
note that `worldgen` currently ignores `fallback`, this should be added to TODOs for feature parity